cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt

```diff
@@ -241,15 +245,23 @@ private class ServiceImpl(
                             ccpaMessageMetaData = it.campaigns?.ccpa?.messageMetaData
                         }
 
-                        if (!campaignManager.hasLocalData) {
-                            it.campaigns?.gdpr?.TCData?.let { tc -> dataStorage.tcData = tc.toMapOfAny() }
+                        if (campaignManager.hasLocalData.not()) {
+
+                            // save tc data in the data storage
+                            it.campaigns?.gdpr?.TCData?.let { tcData ->
+                                dataStorage.tcData = tcData.toMapOfAny()
+                            }
+
+                            // clear out previous and save new gpp data in the data storage
+                            it.campaigns?.ccpa?.gppData?.let { gppData ->
+                                dataStorage.clearGppData()
+                                dataStorage.gppData = gppData.toMapOfAny()
```

Comment:
What you think of adding the `clearGppData()` to the `dataStorage.gppData` setter? This way we don't have to call it always, just by setting `dataStorage.gppData = ...` will clean it for us.

Reply:
@andresilveirah agree, that makes more sense, since we don't have any case when we update `gppData` without clearing out local data first

Reply:
@andresilveirah actually, I think we need to clear out data storage outside of the setter, since what if we already have local gpp data, and the response from the BE doesn't have gpp in it? I assume we would have to clear out the local data, since our BE is the single source of truth in this case

Reply:
If the response doesn't have GPP data in it, we should set it to `null` (i.e. remove all GPP from local storage). So, having the `clearGppData()` in the setter will work just fine.

Reply:
@andresilveirah you were right, I fixed this already. The trick was that we are not able to set `dataStorage.gppData` to `null`, since the value itself is not nullable (if we don't have anything in the `dataStorage` while trying to access the value it will return an empty map). You can see the solution for that in this PR.

Reply:
IMO having `clear` method outside of getter/setter is preferable since it is a separate action and therefore adding it to `get`/`set` action breaks single responsibility principle 🤓

Reply:
@Nevazhnovu that is also a good point, and since in the near future (I assume) we are going to separate `GPP` from the `CPPA`, we will definitely rethink the approach for that

Reply:
I'd usually agree with separating the two things. However, it's part of the business logic that we _always_ want to clear gpp data and _then_ set its new values.
The problem with separating the two functionalities (clearing + set) is that it makes it easier for anyone to forget clearing it first.